src/main/java/org/springframework/data/elasticsearch/core/IndexOperations.java

```diff
@@ -19,44 +19,181 @@
 import java.util.Map;
 
 import org.elasticsearch.cluster.metadata.AliasMetaData;
+import org.springframework.data.elasticsearch.core.document.Document;
 import org.springframework.data.elasticsearch.core.mapping.IndexCoordinates;
 import org.springframework.data.elasticsearch.core.query.AliasQuery;
+import org.springframework.lang.Nullable;
 
 /**
  * The operations for the
  * <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/indices.html">Elasticsearch Index APIs</a>.
- *
+ * <br/>
+ * IndexOperations can be bound to an entity class or an IndexCoordinate (by {@link #indexOps(Class)} or
+ * {@link #indexOps(IndexCoordinates)}), then it is not necessary to pass the class or IndexCoordinate in every call.
+ * 
  * @author Peter-Josef Meisch
  * @author Sascha Woo
  * @since 4.0
  */
 public interface IndexOperations {
 
+	/**
+	 * returns an {@link IndexOperations} bound to the given entity class.
+	 * 
+	 * @param clazz the entity class
+	 * @return IndexOperations;
+	 */
+	IndexOperations indexOps(Class<?> clazz);
```

Comment:
Let's not introduce `indexOps(…)` methods here. They should live only on `ElasticsearchOperations` only. The deprecated methods that do the forwarding should remain on `ElasticsearchOperations`, too.